src/openms/source/ANALYSIS/MAPMATCHING/TransformationModel.cpp

```diff
@@ -60,5 +63,295 @@ namespace OpenMS
+    valid_weights = getValidXWeights();
+    if (params.exists("x_weight") && checkValidWeight(params.getValue("x_weight"), valid_weights) && !data.empty())
+    {
+      std::string x_weight = params.getValue("x_weight");
+      x_weight_ = x_weight;
+      for (size_t i = 0; i < data.size(); ++i)
+      {
+        data[i].first = unWeightDatum(data[i].first,x_weight_);
+      }
+    }
+    else if (params.exists("x_weight") && checkValidWeight(params.getValue("x_weight"), valid_weights))
+    {
+      std::string x_weight = params.getValue("x_weight");
+      x_weight_ = x_weight;
+    }
+    else{
+      std::string x_weight = "";
+      x_weight_ = x_weight;
+    }
+    // unweight y values
+    valid_weights = getValidYWeights();
+    if (params.exists("y_weight") && checkValidWeight(params.getValue("y_weight"), valid_weights) && !data.empty())
+    {
+      std::string y_weight = params.getValue("y_weight");
+      y_weight_ = y_weight;
+      for (size_t i = 0; i < data.size(); ++i)
+      {
+        data[i].second = unWeightDatum(data[i].second,y_weight_);
+      }
+    }    
+    else if (params.exists("y_weight") && checkValidWeight(params.getValue("y_weight"), valid_weights))
+    {
+      std::string y_weight = params.getValue("y_weight");
+      y_weight_ = y_weight;
+    }
+    else{
+      std::string y_weight = "";
+      y_weight_ = y_weight;
+    }
+  }
+
+  bool TransformationModel::checkValidWeight(const std::string& weight, const std::vector<std::string>& valid_weights) const
+  {    
+    bool valid = false;
+    if (std::find(valid_weights.begin(), valid_weights.end(), weight) != valid_weights.end())
+    {
+      valid=true;
+    }
+    else
+    {
+      std::cout << "weight " + weight + " is not supported" << std::endl;
+    }
+    return valid;
+  }
+  
+  std::vector<std::string> TransformationModel::getValidXWeights() const
+  {
+    //std::vector<std::string> valid_weights{"1/x","1/x2","ln(x)",""}; C++ 11
+    std::vector<std::string> valid_weights;
+    valid_weights.push_back("1/x");
+    valid_weights.push_back("1/x2");
+    valid_weights.push_back("ln(x)");
+    valid_weights.push_back("");
+    return valid_weights;
+  }
+  
+  std::vector<std::string> TransformationModel::getValidYWeights() const
+  {
+    // std::vector<std::string> valid_weights{"1/y","1/y2","ln(y)",""}; C++ 11
+    std::vector<std::string> valid_weights;
+    valid_weights.push_back("1/y");
+    valid_weights.push_back("1/y2");
+    valid_weights.push_back("ln(y)");
+    valid_weights.push_back("");
+    return valid_weights;
+  }
+
+  double TransformationModel::weightDatum(const double& datum, const std::string& weight) const
+  { 
+    double datum_weighted = 0;   
+    if (weight == "ln(x)")
+    {
+      if (datum < 10e-5)
+      {
+        datum_weighted = std::log(10e-5);
+      }
+      else
+      {
+        datum_weighted = std::log(datum);
+      }
+    }
+    else if (weight == "ln(y)")
+    {
+      if (datum < 10e-8)
```

Comment:
why do x and y differ in when you take the default weight? e.g. 10e-5 vs 10e-8 ?